sql/catalyst/src/main/java/org/apache/spark/sql/catalog/v2/expressions/Expression.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalog.v2.expressions;
```

Comment:
an issue with using the sql.catalog namespace is that it is not clear this is a "source" or "connector". In general, I think we should have separate API packages for library / data source developers vs end users. We already use sql.catalog for end user facing catalog functionalities.

Reply:
How about `org.apache.spark.sql.logical.expressions` then? It doesn't matter to me.

Reply:
i think it is better to create a namespace for data sources (or connectors). as an example:

org.apache.spark.sql.sources

or

org.apache.spark.sql.connectors

and within it you have 

catalog

and

table ?

Reply:
It sounds like you're suggesting a larger reorganization than just where the expressions go. Let's discuss this at tonight's sync.

Reply:
I like the `org.apache.spark.sql.connectors` packge, but moving classes will cause code conflict for all the DS v2 PRs. Maybe do it latter?

Reply:
When are we going to do this?

Reply:
First, we have to decide on what all we need to move. I don't think that is clear yet. The next concern is getting #24246 in because it is a blocker for a lot of work that can be done in parallel. After that, it would be nice to coordinate to avoid breaking lots of PRs, but that's less of a concern.

So to answer your question, I think we should do this after #24246 and after we've decided what needs to move and what the new organization should be.

The actual changes should be simple and quick to review, but would cause too much delay combined into a PR with other changes.